app/src/main/java/university/innopolis/madgroup3/onboarding/activities/LoginActivity.kt

```diff
@@ -6,36 +6,43 @@ import android.text.TextUtils
 import android.widget.Toast
 import kotlinx.android.synthetic.main.activity_login.*
 import university.innopolis.madgroup3.onboarding.R
+import university.innopolis.madgroup3.onboarding.data.repositories.TokenRepository
+import javax.inject.Inject
 
 class LoginActivity : AppCompatActivity() {
 
+    @Inject lateinit var tokenRepository: TokenRepository
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_login)
 
         login_btn.setOnClickListener {
-            val inputEmail = login_email.text.toString()
+            val inputUsername = login_username.text.toString()
             val inputPw = login_pw.text.toString()
 
-            val errorText = validateCredentials(inputEmail, inputPw)
+            val errorText = validateCredentials(inputUsername, inputPw)
             if (!errorText.isNullOrBlank()) {
                 val errorToast = Toast.makeText(applicationContext, errorText, Toast.LENGTH_SHORT)
                 errorToast.show()
             } else {
-                // TODO: API call with inputEmail and inputPw
-                // TODO: Fire an Intent after successful completion
-                // TODO: Fire a Toast on unsuccessful completion (error)
+                val token = tokenRepository.requestToken(inputUsername, inputPw)
+                token ?: showTokenFailToast()
```

Comment:
Here we need to fail an authentication

Reply:
That was my thinking: if `token` is falsy, then show toast. Did I not do it like that?

Reply:
That's right, but the function won't stop its execution. So you need to add smth like `return show TokenFailToast()`